src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
```

Comment:
Not having a ton of familiarity with the code, my gut votes for 2; it sounds like delete-certificate! is doing double duty and having two explicit functions seems better than one function with implied behavior.